### PR TITLE
(maint) Update beaker to support pupppet5 style installation

### PR DESCRIPTION
**Pull request by melissa**

### CODE REVIEW

**Comment by melissa** [comment]
rspec tests are broken, but I still plan on updating the tests

**Comment by melissa** [comment]
rspec tests are fixed and working, I'd just like to add a few new tests for the new code I've added

**Comment by melissa** [comment]
These changes are only needed to access the public repos, users shouldn't have to do anything special yet to access the packages on builds.delivery.puppetlabs.net.

**Comment by mchllweeks** [comment]
@johnduarte Do you have time to also review this pr?

**Comment by melissa** [comment]
I'd still like to update the tests, I haven't had time to do that yet.

File: spec/beaker-puppet/install_utils/foss_utils_spec.rb [hunk 1]
```diff
@@ -831,6 +837,14 @@ def stub_uninteresting_portions_of_install_puppetlabs_dev_repo!
       }.to raise_error(RuntimeError, /Puppet MSI at http:\/\/downloads.puppetlabs.com\/windows\/puppet-agent-VERSION-x64.msi does not exist!/)
     end
 
+    it 'returns the right link when puppet collection is set' do
+      allow( host ).to receive( :is_x86_64? ).and_return( true )
+      allow( host ).to receive( :system_temp_path ).and_return( 'system_path' )
+
+      subject.install_puppet_agent_from_msi_on( host, opts.merge({:puppet_collection => 'puppet'}) )
+      expect(subject).to receive(:link_exists?).with("#{opts[:win_download_url]}/puppet/puppet-agent-VERSION-x64.msi").and_return( true )
+    end
```
Comment (melissa) [inline spec/beaker-puppet/install_utils/foss_utils_spec.rb]: I'm having issues with these tests. How do I get it to not execute all the things? This is the error I get with the current iteration. If I move line 845 up after 841, it gets to the cygwin check, but it looks like I should just be able to only check the thing getting passed into `:link_exists?`, and end there. What am I missing?

```
  1) ClassMixedWithDSLInstallUtils#install_puppet_agent_from_msi_on returns the right link when puppet collection is set
     Failure/Error: raise "Puppet MSI at #{link} does not exist!"

     RuntimeError:
       Puppet MSI at http://downloads.puppetlabs.com/windows/puppet/puppet-agent-VERSION-x64.msi does not exist!
     # ./lib/beaker-puppet/install_utils/foss_utils.rb:603:in `block in install_a_puppet_msi_on'
     # ./lib/beaker-puppet/install_utils/foss_utils.rb:596:in `install_a_puppet_msi_on'
     # ./lib/beaker-puppet/install_utils/foss_utils.rb:590:in `block in install_puppet_agent_from_msi_on'
     # ./lib/beaker-puppet/install_utils/foss_utils.rb:576:in `install_puppet_agent_from_msi_on'
     # ./spec/beaker-puppet/install_utils/foss_utils_spec.rb:844:in `block (3 levels) in <top (required)>'
```

Comment (melissa) [reply spec/beaker-puppet/install_utils/foss_utils_spec.rb]: For example, I added in a second test in the other way I can think to do this below. Results are:
```
  1) ClassMixedWithDSLInstallUtils#install_puppet_agent_from_msi_on returns the right link when puppet collection is set
     Failure/Error: raise "Puppet MSI at #{link} does not exist!"

     RuntimeError:
       Puppet MSI at http://downloads.puppetlabs.com/windows/puppet/puppet-agent-VERSION-x64.msi does not exist!
     # ./lib/beaker-puppet/install_utils/foss_utils.rb:603:in `block in install_a_puppet_msi_on'
     # ./lib/beaker-puppet/install_utils/foss_utils.rb:596:in `install_a_puppet_msi_on'
     # ./lib/beaker-puppet/install_utils/foss_utils.rb:590:in `block in install_puppet_agent_from_msi_on'
     # ./lib/beaker-puppet/install_utils/foss_utils.rb:576:in `install_puppet_agent_from_msi_on'
     # ./spec/beaker-puppet/install_utils/foss_utils_spec.rb:844:in `block (3 levels) in <top (required)>'

  2) ClassMixedWithDSLInstallUtils#install_puppet_agent_from_msi_on returns the right link when puppet collection is not set
     Failure/Error: if host.is_cygwin?

     NoMethodError:
       undefined method `is_cygwin?' for #<Hash:0x007f40ba102668>
     # ./lib/beaker-puppet/install_utils/foss_utils.rb:609:in `block in install_a_puppet_msi_on'
     # ./lib/beaker-puppet/install_utils/foss_utils.rb:596:in `install_a_puppet_msi_on'
     # ./lib/beaker-puppet/install_utils/foss_utils.rb:590:in `block in install_puppet_agent_from_msi_on'
     # ./lib/beaker-puppet/install_utils/foss_utils.rb:576:in `install_puppet_agent_from_msi_on'
     # ./spec/beaker-puppet/install_utils/foss_utils_spec.rb:853:in `block (3 levels) in <top (required)>'
```

Comment (tvpartytonight) [reply spec/beaker-puppet/install_utils/foss_utils_spec.rb]: @melissa it looks like the method `:install_a_puppet_msi_on` doesn't have any spec tests; if you wanted to continue adding tests from the method `:install_puppet_agent_from_msi_on` you would need to mock out the rest of the method.

I would suggest adding a new `describe` block that tests `:install_a_puppet_msi_on`, with your code that tests it removed to a new method that `:install_a_puppet_msi_on` calls. Maybe call it something like `build_msi_path`?

The code that I would move to that method is:
```` ruby
if opts[:puppet_collection] && opts[:puppet_collection].match(/puppet\d*/)
              link = "#{opts[:win_download_url]}/#{opts[:puppet_collection]}/#{host['dist']}.msi"
            else
              link = "#{opts[:win_download_url]}/#{host['dist']}.msi"
            end
            if not link_exists?( link )
              raise "Puppet MSI at #{link} does not exist!"
            end
````

**Comment by melissa** [comment]
Tests have been updated, thank you @tvpartytonight !

**Comment by melissa** [comment]
@johnduarte @kevpl @tvpartytonight I'd love to get some eyes on this. It sounds like folks are ready to start using this code.

**Comment by melissa** [comment]
@kevpl @tvpartytonight any additional things that need to happen before this can be merged?

**Comment by tvpartytonight** [comment]
👍

**Comment by tvpartytonight** [comment]
@kevpl I don't see any jenkins bot testing comments...I assume I can just kick one off, for safety's sake?

**Comment by tvpartytonight** [comment]
@puppetlabs-jenkins test this please

**Comment by melissa** [comment]
From what I can tell, there are no automated tests that get kicked off on beaker-puppet PRs

**Comment by kevpl** [comment]
They ran when @tvpartytonight kicked them, but I don't think they put comments in the conversation anymore, just a status on the commit itself.

**Pull request merged**